lib/travis/logs/app.rb

```diff
@@ -124,6 +165,64 @@ def initialize(existence = nil, pusher = nil, database = nil, log_part_service =
 
         status 204
       end
+
+      put '/logs/:job_id/archived' do
+        # TODO: de-duplicate auth stuff
+        halt 500, 'authentication token is not set' if ENV['AUTH_TOKEN'].to_s.strip.empty?
+        halt 403 if request.env['HTTP_AUTHORIZATION'] != "token #{ENV['AUTH_TOKEN']}"
+
+        halt 501
```

Comment:
Should we return an error body here too to make it a little more obvious what's going on if you don't recognize the status code right away?